### ISSUE TITLE
Support for PathFormer?

### ISSUE THREAD

**Issue by dyhan316** [thuml/Time-Series-Library#565]
Hello! First I want to send my gratitude to the authors of this code, as it enabled us to test various models on our intracranial EEG data! 

We have a few questions if you don't mind. 
**1. Are you planning on supporting PathFormer?** : We believe that path former, with its adaptive patch size selection mechanism might be optimal for our ECoG dataset, so was wondering!
**2. What is the SOTA model?** : Reading papers brings confusion as papers that use the same model as baseline show different performances, so comparing models that are released at the same year (ex : TimesMixer and Pathformer) is impossible. We know that your project is supposed to remedy this, but we since your project does not have every up-to-date model, we were wondering : **what is the most SOTA model right now ?** 

Thank you in advance for your response!  :)

**Comment by wuhaixu2016** [comment]
Hi, thanks for your valuable suggestion.

(1) We have contacted with the author of Pathformer and they will propose a pull request to include Pathformer to TSlib. Maybe in this month.

(2) You have raised a very important question. 

- I believe that persistently refreshing the results in the long-term benchmarks may be meaningless. Till now, I think most of researchers have noticed that the widely used six datasets have some clear bias. Specifically, their temporal patterns is too simple, where even a simple linear regression model can achieve a fairly good performance. FYI, I still do not believe that a plain linear model is good for prediction. They are just overfitting these long-term benchmarks in a very tricky way.

- Besides, many papers are evaluated in various settings. I noticed that after TimesNet, many papers try to input longer series or search hyperparameters for different datasets or even different prediction lengths, which makes the comparison much more difficult. That is why we test two settings in the TimeMixer paper, including both unified hyperparameter and searched settings.

In conclusion, I think the performance in M4 or the PEMS dataset is much more reliable, which is much larger and contains more diverse patterns. So, my answer will be "Let us see the performance in M4 and PEMS benchmarks".

**Comment by dyhan316** [comment]
@wuhaixu2016  Woah thank you so much for your response!

(1) Great! We'll work on trying out various other models in the mean time. Thank you so much!
(2) Oh I see! Thank you for your insights

Since you are an expert at time series modeling, I have a quick question if you don't mind : do you think that foundational time series models such as MOMENT should also be tested as a baseline, along with the models in your library? 

Thank you :)

**Comment by wuhaixu2016** [comment]
Of course. MOMENT or other "foundational time series models" are evaluated on the same benchmark. I think the comparison among them is natural and convenient. We do not include MOMENT into TSLib just because their training paradigm is different from the conventional one-dataset-train-one-dataset-test setting.

However, I have to emphasize that, we should notice the difference between current "foundational time series models" (so-called) and other time series models (e.g. TimesNet or PatchTST) is just the whether this model has been pre-trained from a large dataset. Thus, to ensure a fair comparison, we should also train TimesNet based on the same pre-training datasets as MOMENT. Otherwise, you cannot get any useful and rigorous conclusions from this comparison.

In addition, I think we are still far away from a true "foundation model". The current capability of foundation ts models is far less than GPT-4. I think most of the current works about "foundation ts models" are just doing something like BERT (new pertaining strategy) or Transformer (new scalable backbone).

**Comment by dyhan316** [comment]
Thank you so much for your response!

Oh I see, not including foundation models makes sense since TSLib is about comparing the architecture themselves! 

Oh I see, yeah it seems that time-series "foundation model" are limited in capabilities, compared to the LLMs. What do you  think would be needed to create a real "foundation model" for time series, other than the new pretraining paradigm and scalable backbone you mentioned? We are trying to create a "foundation model" for human neural data, so your insights would be greatly helpful!
